blog/_src/posts/2018-12-02-java-transient.md

```diff
@@ -0,0 +1,361 @@
+    Title: Java and Migratory Typing
+    Date: 2018-12-02T14:41:53
+    Tags: migratory typing, gradual typing, java, by Ben Greenman
+
+The _transient_ approach to migratory typing (circa [2014](http://homes.sice.indiana.edu/mvitouse/papers/dls14.pdf))
+ is similar to type erasure in Java (circa [2004](https://docs.oracle.com/javase/1.5.0/docs/relnotes/features.html)).
+
+<!-- more -->
+
+## Migratory typing
+
+The goal of _migratory typing_ is to enrich the type system of a language
+ without breaking backwards compatibility.
+Ideally, code that uses the enriched types:
+
+- (G1) benefits from new ahead-of-time checks,
+- (G2) benefits from stronger run-time guarantees, and
+- (G3) may interact with all kinds of existing code.
```

Comment:
Since G1, G2, and G3 are only referenced once (afaict), it may not be worth having a shorthand.